diff -rupN cinnamon-settings-daemon-2.0.7/.pc/90_set_gmenus_xsettings.patch/plugins/xsettings/gsd-xsettings-manager.c cinnamon-settings-daemon-2.0.7_working_on/.pc/90_set_gmenus_xsettings.patch/plugins/xsettings/gsd-xsettings-manager.c
--- cinnamon-settings-daemon-2.0.7/.pc/90_set_gmenus_xsettings.patch/plugins/xsettings/gsd-xsettings-manager.c	2013-11-07 10:29:10.000000000 +0000
+++ cinnamon-settings-daemon-2.0.7_working_on/.pc/90_set_gmenus_xsettings.patch/plugins/xsettings/gsd-xsettings-manager.c	2013-11-15 00:44:02.763907853 +0000
@@ -215,7 +215,7 @@
  * - ajax
  * 
  */
-#define DPI_FALLBACK 96
+#define DPI_FALLBACK 142
 
 typedef struct _TranslationEntry TranslationEntry;
 typedef void (* TranslationFunc) (GnomeXSettingsManager *manager,
diff -rupN cinnamon-settings-daemon-2.0.7/plugins/xsettings/csd-xsettings-manager.c cinnamon-settings-daemon-2.0.7_working_on/plugins/xsettings/csd-xsettings-manager.c
--- cinnamon-settings-daemon-2.0.7/plugins/xsettings/csd-xsettings-manager.c	2013-11-07 10:29:10.000000000 +0000
+++ cinnamon-settings-daemon-2.0.7_working_on/plugins/xsettings/csd-xsettings-manager.c	2013-11-15 01:11:12.943872221 +0000
@@ -215,7 +215,7 @@
  * - ajax
  * 
  */
-#define DPI_FALLBACK 96
+#define DPI_FALLBACK 142
 
 typedef struct _TranslationEntry TranslationEntry;
 typedef void (* TranslationFunc) (CinnamonSettingsXSettingsManager *manager,
